### PR TITLE
Update ClassView layout and styles (1/3 for rewriting ClassView)

### PR DESCRIPTION
**Pull request by kungpaogao**
### Summary 

This pull request is the first step towards rewriting the `ClassView` page. This PR deals with the layout and styling of `ClassView`.

- created new component called `ClassViewNew` to house changes for now
- removed lots of nesting of components and styles (e.g., `ClassView` directly imports and uses `Form` component)
  - this allows for better layout management - breakpoints and scaling can be set in parent CSS
- redid `Gauge` layout and styling again to make alignment easier
- fixed lots of small visual glitches (e.g., alignment of scroll bar on right side of course reviews, gauge container right-side alignment)
- updated overflow behavior to be more predictable
- removed `CourseCard` component because it's useless
- see **Breaking Changes** section below for more
- more context in Notion: https://www.notion.so/Rewrite-ClassView-component-76aabbe867194c2387f592279f98017e

### Test Plan 

Tested on macOS: Firefox Developer Edition, Safari, Chrome. Tested on iOS: Safari.

macOS
| old | new |
| - | - |
| image | image |

iOS
| old | new |
| - | - |
| image | image |
- top bar doesn't collapse when scrolling in new implementation (see **Breaking Changes**)
- missing scrollbar
- missing "Leave a Review" button

### Linter Warnings 

n/a

### Notes 

- DO NOT MERGE this into `master` branch - this fix is foundational work for following PRs that will be merged into `kungpaogao/class-view-rewrite`, which will then be merged into `master` as a package
- all this work will probably be rebased onto @jack-farley's Bootstrap 4 work
- we should rethink spacing and how to handle it consistently - right now, I am using pretty arbitrary numbers like `70px` and `35px` based on previous work, but this should be standardized
  - maybe we should look into using CSS variables?? 

### Breaking Changes  

- this current PR is missing several features
  - missing ability to add review on mobile - no buttons or form
  - removed `ResultsDisplayMobile` (for now)
  - missing scroll interactions (e.g., collapsing top bar on mobile)
  - missing past review count and sorting feature for reviews
- some bugs
  - fixed nested `.container` and `.row` usage in `index.tsx` and children components - this causes the `ResultsDisplay` header to shift left because of negative margins
  - `Form` component doesn't scale super well at the moment, but that will be addressed in future PRs (probably 3/3)

### CODE REVIEW

File: client/src/ui/ClassViewNew.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                  label="Difficulty"
+                />
+              </div>
+              <div className={styles.gauge}>
+                <Gauge rating={selectedClass.classWorkload} label="Workload" />
+              </div>
+            </div>
+            <h2 className={styles.pastReviews}>Past Reviews</h2>
+            <div className={styles.courseReviews}>
+              <CourseReviews
```
Comment (jack-farley) [inline client/src/ui/ClassViewNew.tsx:102]: I'm seeing a double scroll bar in the past reviews section, not sure why that is happening.

File: client/src/ui/ClassViewNew.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import Form from "./Form";
+import { Class } from "common";
+
+enum PageStatus {
+  Loading,
+  Success,
+  Error,
+}
+
+export default function ClassView() {
```
Comment (jack-farley) [inline client/src/ui/ClassViewNew.tsx:19]: Great job on rewriting class view - this looks much cleaner!

File: client/src/ui/ClassViewNew.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        <h2>Sorry, we couldn't find the class you're searching for.</h2>
+        <h2>Please search for a different class.</h2>
+      </div>
+    );
+  }
+
+  if (pageStatus === PageStatus.Success && !!selectedClass) {
+    courseVisited(selectedClass?.classSub, selectedClass?.classNum);
+    return (
+      <div className={`row ${styles.classView}`}>
```
Comment (jack-farley) [inline client/src/ui/ClassViewNew.tsx:64]: I like the switch to css modules - seems like this is a good move for the frontend

**Review by jack-farley — CHANGES_REQUESTED** [review]
This looks pretty good to me. Aside from the stuff in breaking changes, the double scroll bar in past reviews is the only thing I saw that might need changes

**Review by AkashAryal — CHANGES_REQUESTED** [review]
Wow this looks like it was a lot of work! Good job Andrew! The only things I see is that on windows/ubuntu there are scroll bars in the class view as noted by Jack.

**Comment by jack-farley** [comment]
Not seeing the double scrollbar anymore on windows - good job!

**Review by AkashAryal — APPROVED** [review]
The double scroll bars dont show up in ubuntu anymore. Nice!

**Review by jack-farley — APPROVED** [review]
Looks good!

**Pull request merged**